src/GraphTopology.hpp

```diff
@@ -144,28 +107,22 @@ class GraphTopology
     mfem::SparseMatrix Agg_edge_;
     mfem::SparseMatrix Agg_vertex_;
     mfem::SparseMatrix face_Agg_;
-    mfem::SparseMatrix Agg_face_;
     mfem::SparseMatrix face_edge_;
     ///@}
 
-    /// "face" to boundary attribute table
-    mfem::SparseMatrix face_bdratt_;
+    /// pointer to coarse graph
+    const Graph* coarse_graph_;
```

Comment:
Two things: (1) can we make this a `shared_ptr`? (2) does the pointer itself have to be public?

Reply:
(1) yes, `coarse_graph_ ` should be a `shared_ptr` here.
(2) the only place we need this public access is in the test `lineargraph`, where we manually build the coarse graph and set it in `GraphTopology`. So if `coarse_graph_` is not public, we should make a "Setter" for it. I will make a change for this alternative.